src/main/java/io/fabric8/maven/docker/StartMojo.java

```diff
@@ -483,6 +488,45 @@ private String getTcpHost(WaitConfiguration.TcpConfiguration tcpConfig, Properti
         return host;
     }
 
+    private WaitUtil.WaitChecker getHealthyWaitChecker(final String imageConfigDesc, final DockerAccess docker, final String containerId, final List<String> logOut) {
+
+
+        return new WaitUtil.WaitChecker() {
+
+            private boolean first = true;
+
+            @Override
+            public boolean check() {
+                try {
+                    final InspectedContainer container = docker.getContainer(containerId);
+                    if (container == null) {
+                        log.debug("HealthyWaitChecker:  Container %s not found");
+                        return false;
+                    }
+
+                    if (first) {
+                        final String healthcheckTests = container.healthcheckTests();
+                        if (healthcheckTests == null) {
+                            throw new IllegalArgumentException("Can not wait for healthy state of "+imageConfigDesc+". No HEALTHCHECK configured.");
+                        }
+                        log.info("%s: Waiting on healthcheck: '%s'", imageConfigDesc, healthcheckTests);
```

Comment:
I would only use a short, fixed message for `logOut` and post the full health-check with `log.verbose` (once). That is in line with how Dockerfile instruction are only printed with verbose level when an image is build.